packages/cli-helpers/src/auth/__tests__/authFiles.test.ts

```diff
@@ -1,13 +1,14 @@
 // Have to use `var` here to avoid "Temporal Dead Zone" issues
 let mockBasePath = ''
-let mockIsTypeScriptProject = true
 globalThis.__dirname = __dirname
 
-jest.mock('../../lib/paths', () => {
+vi.mock('../../lib/paths', async (importOriginal) => {
   const path = require('path')
+  // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+  const orginalPaths = await importOriginal<typeof import('../../lib/paths')>()
```

Comment:
This looks interesting! Can you explain what's going on here?

Reply:
This was taken from the `vi.mock` docs: https://vitest.dev/api/vi.html#vi-mock

Having the type of the original module is of course useful. There was one case where we were accessing a property on the original mock that didn't actually exist and that was now clear when it had the correct types.

The ESLint rule is a small pain since it appears not to like that particular syntax. I haven't tried any variations that might get around it.